scripts/ci/check_compliance.py

```diff
@@ -59,6 +59,10 @@ def get_files(filter=None, paths=None):
     paths_arg = ('--', *paths) if paths else ()
     out = git('diff', '--name-only', *filter_arg, COMMIT_RANGE, *paths_arg)
     files = out.splitlines()
+    for file in files:
+        if not os.path.isfile(os.path.join(GIT_TOP, file)):
+            # Drop submodule directories from the list.
+            file.remove(file)
```

Comment:
Shouldn't this be `files.remove(file)`? Also, is this safe while iterating?

Reply:
hey good points, yes it should be files, changed, and no apparently it's not safe, added an extra `list()` to make a copy (https://sopython.com/canon/95/removing-items-from-a-list-while-iterating-over-the-list/). Thanks for spotting these.